### PR TITLE
[docs] add graph_asset to api docs

### PR DESCRIPTION
**Pull request by jamiedemaria**
## How I Tested These Changes
in the vercel preview i went to the graph backed assets concept page and clicked `graph_asset` in the relevant apis box. Confirmed that it directed me to the `graph_asset` api docs

### CODE REVIEW

**Comment by jamiedemaria** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13084** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13084" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13084?utm_source=stack-comment).

**Review by erinkcochran87 — COMMENTED** [review]
I left a few comments on the Vercel preview, since the relevant code for the API content isn't in this PR: https://dagster-git-jamie-graphassetapidocs-elementl.vercel.app/_apidocs/assets#dagster.graph_asset

File: python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py [hunk 1]
```diff
@@ -664,12 +664,12 @@ def graph_asset(
 
     Args:
         name (Optional[str]): The name of the asset.  If not provided, defaults to the name of the
-            decorated function. The asset's name must be a valid name in dagster (ie only contains
-            letters, numbers, and _) and may not contain python reserved keywords.
+            decorated function. The asset's name must be a valid name in Dagster (ie only contains
+            letters, numbers, and underscore) and may not contain Python reserved keywords.
```
Comment (erinkcochran87) [inline python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py:668]: Nit - this should be plural, so `underscores`

File: python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py [hunk 2]
```diff
@@ -664,12 +664,12 @@ def graph_asset(
...
         key_prefix (Optional[Union[str, Sequence[str]]]): If provided, the asset's key is the
             concatenation of the key_prefix and the asset's name, which defaults to the name of
-            the decorated function. Each item in key_prefix must be a valid name in dagster (ie only
-            contains letters, numbers, and _) and may not contain python reserved keywords.
+            the decorated function. Each item in key_prefix must be a valid name in Dagster (ie only
+            contains letters, numbers, and underscore) and may not contain Python reserved keywords.
```
Comment (erinkcochran87) [inline python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py:672]: Same here, re: `underscores`

**Review by erinkcochran87 — APPROVED** [review]
Approving, but pls see comments before merging :) Small grammar issue but otherwise this looks good!

File: python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py (hunk 1 above)
Comment (jamiedemaria) [reply python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py:668]: not sure how much this matters, but i noticed that the docs for `@asset` say "letters, numbers, and _)". Should i update that to "letters, numbers, and underscores) for consistency? I wonder if there are other places in our docs that say the same thing

**Pull request merged**